processing/src/main/java/org/apache/druid/query/groupby/GroupByQuery.java

```diff
@@ -313,9 +319,9 @@ public LimitSpec getLimitSpec()
     return limitSpec;
   }
 
-  @JsonInclude(JsonInclude.Include.NON_NULL)
-  @JsonProperty("subtotalsSpec")
   @Nullable
+  @JsonProperty("subtotalsSpec")
+  @JsonInclude(JsonInclude.Include.NON_EMPTY)
```

Comment:
IIRC empty `subtotalsSpec` is treated differently from null `subtotalsSpec`. If that's accurate then this should be `NON_NULL`.